src/checkers/inference/solver/backend/maxsat/encoder/MaxSATSubtypeConstraintEncoder.java

```diff
@@ -8,25 +8,73 @@
 import checkers.inference.solver.frontend.Lattice;
 import checkers.inference.util.ConstraintVerifier;
 import org.checkerframework.javacutil.AnnotationUtils;
-import org.sat4j.core.Vec;
 import org.sat4j.core.VecInt;
 
 import javax.lang.model.element.AnnotationMirror;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-/**
- * Created by mier on 06/11/17.
- */
-public class MaxSATSubtypeConstraintEncoder extends MaxSATAbstractBinaryConstraintEncoder implements SubtypeConstraintEncoder<VecInt[]>{
+public class MaxSATSubtypeConstraintEncoder extends MaxSATAbstractConstraintEncoder implements SubtypeConstraintEncoder<VecInt[]> {
 
     public MaxSATSubtypeConstraintEncoder(Lattice lattice, ConstraintVerifier verifier, Map<AnnotationMirror, Integer> typeToInt) {
         super(lattice, verifier, typeToInt);
     }
 
+    /**
+     * for subtype constraint, if supertype is constant slot, then the subtype
+     * cannot be the super type of supertype, same for subtype
+     *
+     * @param mustNotBe
```

Comment:
If we don't have an explanation of a `@param`, remove it. (Or a better way would be add a simple explanation of this `@param`. The same as `@return`.